moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_calcs.h

```diff
@@ -77,30 +83,44 @@ class JogCalcs
 
   sensor_msgs::JointState incoming_joints_;
 
+  /** \brief Do jogging calculations for Cartesian twist commands. */
   bool cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, JogArmShared& shared_variables, std::mutex& mutex);
 
-  bool jointJogCalcs(const control_msgs::JointJog& cmd, JogArmShared& shared_variables);
+  /** \brief Do jogging calculations for direct commands to a joint. */
+  bool jointJogCalcs(const control_msgs::JointJog& cmd, JogArmShared& shared_variables, std::mutex& mutex);
 
-  // Parse the incoming joint msg for the joints of our MoveGroup
+  /** \brief Update the stashed status so it can be retrieved asynchronously */
+  void updateCachedStatus(JogArmShared& shared_variables, std::mutex& mutex);
+
+  /** \brief Parse the incoming joint msg for the joints of our MoveGroup */
   bool updateJoints(std::mutex& mutex, const JogArmShared& shared_variables);
 
+  /** \brief If incoming velocity commands are from a unitless joystick, scale them to physical units.
+   * Also, multiply by timestep to calculate a position change.
+   */
   Eigen::VectorXd scaleCartesianCommand(const geometry_msgs::TwistStamped& command) const;
 
+  /** \brief If incoming velocity commands are from a unitless joystick, scale them to physical units.
+   * Also, multiply by timestep to calculate a position change.
+   */
   Eigen::VectorXd scaleJointCommand(const control_msgs::JointJog& command) const;
 
   bool addJointIncrements(sensor_msgs::JointState& output, const Eigen::VectorXd& increments) const;
 
-  // Suddenly halt for a joint limit or other critical issue.
-  // Is handled differently for position vs. velocity control.
+  /** \brief Suddenly halt for a joint limit or other critical issue.
+   * Is handled differently for position vs. velocity control.
+   */
   void suddenHalt(trajectory_msgs::JointTrajectory& joint_traj);
 
-  void publishWarning(bool active) const;
+  /** \brief Publish the status of the jogger to a ROS topic */
+  void publishStatus(StatusCode status) const;
```

Comment:
Looks like this is always used with the class variable `status_`. Why not do this?
```suggestion
  void publishStatus() const;
```

Reply:
done